mne/preprocessing/maxwell.py

```diff
@@ -0,0 +1,277 @@
+        with x-axis.
+    pol : float
+        Polar (or colatitudinal) spherical coordinate [0, pi]. 0 is aligned
+        with z-axis.
+
+    Returns
+    -------
+    y : complex float
+        The spherical harmonic value at the specified azimuth and polar angles
+    """
+    assert np.abs(order) <= degree, ('Absolute value of expansion coefficient'
+                                     ' must be <= degree')
+
+    # Get function for Legendre derivatives
+    #lut, n_fact = _get_legen_table('meg', False, 100)
+    #lut_deriv_fun = partial(_get_legen_lut_accurate, lut=lut)
+
+    #TODO: Decide on notation for spherical coords
+    #TODO: Should factorial function use floating or long precision?
+    #TODO: Check that [-m to m] convention is correct for all equations
+
+    #Ensure that polar and azimuth angles are arrays
+    polar = np.array(pol)
+    azimuth = np.array(az)
+
+    # Real valued spherical expansion as given by Wikso, (11)
+    base = np.sqrt(((2 * degree + 1) / (4 * np.pi) *
+                    factorial(degree - order) / factorial(degree + order)) *
+                   lpmv(order, degree, np.cos(polar)))
+    if order < 0:
+        return base * np.sin(order * azimuth)
+    else:
+        return base * np.cos(order * azimuth)
+
+    #TODO: Check speed of taking real part of scipy's sph harmonic function
+    # Note reversal in notation order between scipy and original SSS papers
+    # Degree/order and theta/phi reversed
+    #return np.real(sph_harm(order, degree, az, pol))
+
+
+def get_num_harmonics(in_order, out_order):
+    """
+    Compute total number of spherical harmonics.
+
+    Parameters
+    ---------
+    in_order : int
+        Internal expansion order
+    out_order : int
+        External expansion order
+
+    Returns
+    -------
+    M : int
+        Total number of spherical harmonics
+    """
+
+    #TODO: Eventually, reuse code in field_interpolation
+
+    return in_order ** 2 + 2 * in_order + out_order ** 2 + 2 * out_order
+
+
+def _alegendre_deriv(degree, order, val):
+    """
+    Compute the derivative of the associated Legendre polynomial at a value.
+
+    Parameters
+    ----------
+    degree : int
+        Degree of spherical harmonic. (Usually) corresponds to 'l'
+    order : int
+        Order of spherical harmonic. (Usually) corresponds to 'm'
+    val : float
+        Value to evaluate the derivative at
+
+    Returns
+    -------
+    dPlm
+        Associated Legendre function derivative
+    """
+
+    #TODO: Eventually, probably want to switch to look up table but optimize
+    # later
+    C = 1
+    if order < 0:
+        order = abs(order)
+        C = (-1) ** order * fact(degree - order) / fact(1 + order)
+    return C * (order * val * lpmv(order, degree, val) + (degree + order) *
+                (degree - order + 1) * np.sqrt(1 - val ** 2) *
+                lpmv(order, degree - 1, val)) / (1 - val ** 2)
+
+
+def _grad_in_components(degree, order, rad, az, pol, lut_func):
```

Comment:
we should probably standardize `lut_fun`. In the field interpolation code, it is inconsistent. Somtimes, it is called `lut` and sometimes `lut_fun`. And here I see `lut_func`